capstone/src/main/java/com/google/sps/servlets/BlackOwnedRestaurantsDataServlet.java

```diff
@@ -14,82 +14,107 @@
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.FetchOptions.Builder;
+import java.util.Scanner;
 
 /** Returns black owned restaurants data as a JSON object */
-@WebServlet("/black-owned-restaurants-data")
+@WebServlet("/black-owned-restaurants")
 public class BlackOwnedRestaurantsDataServlet extends HttpServlet {
+  private static final String DATABASE_NAME = "BlackOwnedRestuarants";
 
-  private ArrayList<String> blackRestaurants = new ArrayList<>();
-  private ArrayList<PlaceDetails> detailedPlaces = new ArrayList<>();
-  
   private RestaurantDetailsGetter details = new RestaurantDetailsGetter();
+  private RestaurantQueryHelper queryHelper = new RestaurantQueryHelper();
+
+/** scrapes business names from source */
+  private ArrayList<String> getRestaurantNames() throws IOException{
+    ArrayList<String> blackOwnedRestaurants = new ArrayList<>();
 
-  @Override
-  public void init() {
     Scanner scanner =
         new Scanner(
             getServletContext().getResourceAsStream("/WEB-INF/black-owned-restaurants.csv"));
-    int i = 0;
+
     while (scanner.hasNextLine()) {
-      if (i >= 1) break;
       String line = scanner.nextLine();
       String[] cells = line.split(",");
 
       String name = String.valueOf(cells[0]);
-      detailedPlaces.add(details.request(name));
 
-      blackRestaurants.add(name);
-      i++;
+      blackOwnedRestaurants.add(name);
     }
     scanner.close();
+    return blackOwnedRestaurants;
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String keywords = (String) request.getParameter("keyword");
+    Query query = new Query(DATABASE_NAME).addSort("rating", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    List<Entity> allRestaurants = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+
+    ArrayList<Restaurant> result = new ArrayList<>();
+    for (Entity RestaurantEntity : allRestaurants){
+      if (queryHelper.restaurantContainsKeyword(RestaurantEntity, keywords)) {
+        Restaurant restaurant = queryHelper.makeRestaurantObject(RestaurantEntity);
+        result.add(restaurant);
+        
+        if (result.size() >= queryHelper.MAX_RESULTS) break;
+      }
+    }
 
-    response.setContentType("application/json");
     Gson gson = new Gson();
-    String json = gson.toJson(detailedPlaces);
-    response.getWriter().println(json);
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(result));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Is this how the databases get filled? A post triggers the call to the source data?
A) Could we add a comment to the function saying that's what POST does?
B) This might change with Andrew's work to make a cron job to fetch data